src/Microsoft.SymbolStore/SymbolStores/CacheSymbolStore.cs

```diff
@@ -15,7 +15,13 @@ public CacheSymbolStore(ITracer tracer, SymbolStore backingStore, string cacheDi
             : base(tracer, backingStore)
         {
             CacheDirectory = cacheDirectory ?? throw new ArgumentNullException(nameof(cacheDirectory));
-            Directory.CreateDirectory(cacheDirectory);
+            try
+            {
+                Directory.CreateDirectory(cacheDirectory);
```

Comment:
Could we skip trying to create the directory entirely at this point in the code? It appears the code below will lazily create it when we download the first file. If the directory needs to exist eagerly then we probably shouldn't ignore the exceptions that occur when creation fails.

Reply:
I don't remember if there were any reasons to eagerly create the base cache directory and your right that the below CreateDirectory should create all the directories in the path.  

I'll remove it from the constructor and do some testing.